plugins/modules/synchronize.py

```diff
@@ -412,13 +429,14 @@ def main():
             set_remote_user=dict(type='bool', default=True),
             rsync_timeout=dict(type='int', default=0),
             rsync_opts=dict(type='list', default=[], elements='str'),
-            ssh_args=dict(type='str'),
+            _ssh_args=dict(type='str'),
+            use_ssh_args=dict(type='bool', default=False),
             ssh_connection_multiplexing=dict(type='bool', default=False),
             partial=dict(type='bool', default=False),
             verify_host=dict(type='bool', default=False),
             delay_updates=dict(type='bool', default=True),
             mode=dict(type='str', default='push', choices=['pull', 'push']),
-            link_dest=dict(type='list', elements='str'),
+            link_dest=dict(type='list', elements='path'),
```

Comment:
Changing the type from `str` to `path` is changing behavior (since now `AnsibleModule` does certain modifications to it). Shouldn't this be mentioned in the changelog fragment?

Also the original code explicitly **only** did these transformation in special cases:
```
    # expand the paths
    if '@' not in source:
        source = os.path.expanduser(source)
    if '@' not in dest:
        dest = os.path.expanduser(dest)
```
The code now does these transformations unconditionally. I think this is a bug and should be reverted.

Reply:
Yes! Good catch.  👍